main.js

```diff
@@ -55,103 +58,227 @@ function setupInitialPage() {
-    ],
-  };
-  // Create the event card element
-  const eventCard = document.createElement('div');
-  eventCard.classList.add('event-card'); 
-  // Create the event content markup
-  const contentMarkup = `
-    <div class="container m-12">
-    <div class="card">
-        <div class="event-info">
-            <h2 class="event-name">${eventData.name}</h2>
-            <p class="event-description flex"><img src="./src/assets/description.png" class="img-icon" alt="Description Icon">${eventData.description}</p>
-            <p class="event-location flex"><img src="./src/assets/location.png" class="img-icon" alt="Location Icon">${eventData.location}</p>
-            <p class="event-ticket-categories flex"><img src="./src/assets/ticket.png" class="img-icon" alt="Ticket Icon">${eventData.ticketCategories.map((category) => category.description).join(', ')}</p>
-            <button class="order-button">Order</button>
-        </div>
-        <img src=${eventData.img} alt="img" class="event-image">
-    </div>
-</div>
-  `;
 
-  eventCard.innerHTML = contentMarkup;
-  const eventsContainer = document.querySelector('.events');
-  // Append the event card to the events container
-  eventsContainer.appendChild(eventCard);
+  
+  
+  fetchTicketEvents().then((data) => {
+    addEvents(data);
+  });
 }
-*/
 
-function renderHomePage() {
-  const mainContentDiv = document.querySelector('.main-content-component');
-  mainContentDiv.innerHTML = getHomePageTemplate();
+async function fetchTicketEvents() {
+  const response = await fetch('http://localhost:8080/getAllEvents');
+  const data = await response.json();
+  return data;
 
-  
-  fetch('http://localhost:8080/getAllEvents') // Replace with your actual API endpoint
-    .then(response => response.json())
-    .then(data => {
-      console.log('Received data from the backend:', data);
-      const eventsContainer = document.querySelector('.events');
-      data.forEach(eventData => {
-        const eventCard = createEventCard(eventData);
-        eventsContainer.appendChild(eventCard);
-      });
-    })
-    .catch(error => {
-      console.error('Error fetching event data:', error);
+}
+
+const addEvents = (events) => {
+  const eventsContainer = document.querySelector('.events');
+  eventsContainer.innerHTML = 'No events found.';
+
+  if(events.length) {
+    events.forEach(event => {
+      const eventCard = createEvent(event);
+      eventsContainer.appendChild(eventCard);
     });
+  }
+
+}
+
+const createEvent= (eventData) => {
+  const title=kebabcase(eventData.eventName);
+  
+  const eventElement = createEventElement(eventData,title);
+  return eventElement;
 }
 
-function createEventCard(eventData) {
-  const eventCard = document.createElement('div');
-  eventCard.classList.add('event-card');
-  const ticketCategoriesMarkup = eventData.ticketCategoriesDTO.map(category => category.descriptionEventCategory).join(', ');
+const createEventElement = (event,title) => {
```

Comment:
This function is way to big and has multiple responsibilities.
Consider breaking it down into multiple functions, for example: 
-Functions for creating and handling the quantity buttons and events.
-Functions for creating and handling the checkout button(for the checkout you have an function that handles its event, just add a function that creates and adds the element to the event card) 
-Any other functions that may have a specific role.